x-pack/plugins/infra/server/services/log_entries/log_entry_search_strategy.ts

```diff
@@ -0,0 +1,124 @@
+} from '../../utils/typed_search_strategy';
+import { createGetLogEntryQuery, getLogEntryResponseRT, LogEntryHit } from './queries/log_entry';
+
+type LogEntrySearchRequest = IKibanaSearchRequest<LogEntrySearchRequestParams>;
+type LogEntrySearchResponse = IKibanaSearchResponse<LogEntrySearchResponsePayload>;
+
+export const logEntrySearchStrategyProvider = ({
+  data,
+  sources,
+}: {
+  data: DataPluginStart;
+  sources: IInfraSources;
+}): ISearchStrategy<LogEntrySearchRequest, LogEntrySearchResponse> => {
+  const esSearchStrategy = data.search.getSearchStrategy('ese');
+
+  return {
+    search: (rawRequest, options, dependencies) =>
+      defer(() => {
+        const request = decodeOrThrow(asyncRequestRT)(rawRequest);
+
+        const sourceConfiguration$ = defer(() =>
+          sources.getSourceConfiguration(dependencies.savedObjectsClient, request.params.sourceId)
+        ).pipe(shareReplay(1));
+
+        const recoveredRequest$ = of(request).pipe(
+          filter(asyncGetRequestRT.is),
+          map(({ id: { esRequestId } }) => ({ id: esRequestId }))
+        );
+
+        const initialRequest$ = of(request).pipe(
+          filter(asyncSubmitRequestRT.is),
+          concatMap(({ params }) =>
+            sourceConfiguration$.pipe(
+              map(
+                ({ configuration }): IEsSearchRequest => ({
+                  params: createGetLogEntryQuery(
+                    configuration.logAlias,
+                    params.logEntryId,
+                    configuration.fields.timestamp,
+                    configuration.fields.tiebreaker
+                  ),
+                })
+              )
+            )
+          )
+        );
+
+        return concat(recoveredRequest$, initialRequest$).pipe(
+          take(1),
+          concatMap((esRequest) => esSearchStrategy.search(esRequest, options, dependencies)),
+          map((esResponse) => ({
+            ...esResponse,
+            rawResponse: decodeOrThrow(getLogEntryResponseRT)(esResponse.rawResponse),
+          })),
+          map((esResponse) => ({
+            ...esResponse,
+            ...(esResponse.id
+              ? { id: logEntrySearchRequestStateRT.encode({ esRequestId: esResponse.id }) }
+              : {}),
+            rawResponse: logEntrySearchResponsePayloadRT.encode({
+              data: esResponse.rawResponse.hits.hits.map(createLogEntryFromHit)[0] ?? null,
+              errors: (esResponse.rawResponse._shards.failures ?? []).map(
+                createErrorFromShardFailure
+              ),
+            }),
+          }))
+        );
+      }),
+    cancel: async (id, options, dependencies) => {
+      const { esRequestId } = decodeOrThrow(logEntrySearchRequestStateRT)(id);
+      return await esSearchStrategy.cancel?.(esRequestId, options, dependencies);
+    },
+  };
+};
+
+// exported for tests
+export const logEntrySearchRequestStateRT = rt.string.pipe(jsonFromBase64StringRT).pipe(
+  rt.type({
+    esRequestId: rt.string,
```

Comment:
:information_source: This could just be a string for this simple search strategy, but we'll need to track multiple ids in others. So I opted for consistency here.